core/src/main/java/com/netflix/conductor/core/events/EventProcessor.java

```diff
@@ -1,235 +1,258 @@
+    }
+
+    @SuppressWarnings({"unchecked", "ToArrayCallWithZeroLengthArrayArgument"})
+    private void handle(ObservableQueue queue, Message msg) {
+        try {
+            executionService.addMessage(queue.getName(), msg);
+
+            String event = queue.getType() + ":" + queue.getName();
+            logger.debug("Evaluating message: {} for event: {}", msg.getId(), event);
+            List<EventExecution> transientFailures = executeEvent(event, msg);
+
+            if (transientFailures.isEmpty()) {
+                queue.ack(Collections.singletonList(msg));
+            } else if (queue.rePublishIfNoAck()) {
+                // re-submit this message to the queue, to be retried later
+                // This is needed for queues with no unack timeout, since messages are removed from the queue
+                queue.publish(Collections.singletonList(msg));
+            }
+        } catch (Exception e) {
+            logger.error("Error handling message: {} on queue:{}", msg, queue.getName(), e);
+        }
+    }
+
+    @SuppressWarnings("Guava")
+    @VisibleForTesting
+    Pair<EventExecution, Boolean> execute(EventExecution eventExecution, Action action, String payload) {
+        try {
+            String methodName = "executeEventAction";
+            String description = String.format("Executing action: %s for event: %s with messageId: %s with payload: %s", action.getAction(), eventExecution.getId(), eventExecution.getMessageId(), payload);
+            logger.debug(description);
+
+            Map<String, Object> output = new RetryUtil<Map<String, Object>>().retryOnException(() -> actionProcessor.execute(action, payload, eventExecution.getEvent(), eventExecution.getMessageId()),
+                    this::isTransientException, null, RETRY_COUNT, description, methodName);
+            if (output != null) {
+                eventExecution.getOutput().putAll(output);
+            }
+            eventExecution.setStatus(Status.COMPLETED);
+        } catch (RuntimeException e) {
+            logger.error("Error executing action: {} for event: {} with messageId: {}", action.getAction(), eventExecution.getEvent(), eventExecution.getMessageId(), e);
+            if (e.getCause() instanceof UnsupportedOperationException || (e.getCause() instanceof ApplicationException && ((ApplicationException) e.getCause()).getCode() != ApplicationException.Code.BACKEND_ERROR)) {
+                // not a transient error, fail the event execution
+                eventExecution.setStatus(Status.FAILED);
+                eventExecution.getOutput().put("exception", e.getMessage());
+            } else {
+                // transient error, can be retried
+                return new Pair<>(eventExecution, false);
+            }
+        }
+        return new Pair<>(eventExecution, true);
+    }
+
+    private List<EventExecution> executeEvent(String event, Message msg) throws Exception {
```

Comment:
Consider breaking this method, it has for loops inside for loops, increases the complexity of the code.
Lot of things happening in this method try adding a javadoc even though it is a private method.